tenets.md

```diff
@@ -0,0 +1,423 @@
+
+Similarly, reviewers or maintainers
+SHOULD be able to understand
+how the implementation satisfies the requirements.
+
+Being able to easily compose the ESDK parts
+to satisfy all of your requirements is not enough.
+We expect that the complete list of requirements
+to have conflicting security needs for different types of data.
+A single component that satisfies
+multiple conflicting requirements
+may be complete and expressive,
+but it is also confusing
+and is not [hard to misuse](#Hard-to-misuse).
+
+An assembled components
+should do one thing well.
+Leaving the determination of
+what requirement is being satisfied
+to dynamic runtime behavior is suboptimal.
+
+## Clear Inputs
+
+## Summary
+
+Functions take arguments.
+What these arguments are
+and how they are intended to be used
+SHOULD be clear at the call site.
+This is a specific use of
+[correct by construction](#Correct-by-construction)
+and [easy to use](#easy-to-use).
+
+Code is read much more often
+than it is written
+so we prioritize the expression of intent
+for those who come after us.
+By structuring clarity at the call site
+we maintain the readability of our codebase.
+By structuring our API
+to encourage this same kind of clarity
+we help customers express their intent
+when using our tools.
+
+Examples of features used to express clear inputs
+at the call site are
+named parameters, enums, and option hash.
+
+## Motivation
+
+Though expressing intent at the call site
+makes the AWS Encryption SDK (ESDK) more verbose
+it makes function calls obviously correct by construction.
+
+As a developer gets more experience
+with a library they will remember
+what parameters do what and how they are passed.
+But this requires time and effort.
+
+Due to the critical nature
+of the data handled by the AWS Encryption SDK
+this extensive experience MUST NOT
+be a requirement for review.
+The reviewer, auditor,
+or a later developer performing maintenance
+SHOULD be able to easily understand
+the intent of the parameters that are passed.
+
+This is important because
+while the complexity of systems generally increase,
+there is no requirement that
+the experience and understanding
+of the people maintaining these system must increase.
+Just because something is crystal clear to you today
+does not mean it will be equally clear to you later.
+Let alone someone without your experience and understanding.
+
+We express this in the specification
+by naming options and encouraging enums.
+But also our examples MUST use
+expressive variable names
+to help drive home the intention.
+
+## Sensible Defaults
+
+## Summary
+
+Functions need parameters to know what actions to take.
+Sometimes these parameters should have defaults.
+A parameter that has a default
+will take a value specific if no value is passed.
+
+This simplifies using the function
+because callers do not need
+to select a value.
+
+An example is the principle of least surprise.
+Customers MUST have the freedom consider,
```

Comment:
Fragment, not sure what this is implying.

Reply:
edited